src/browser/BLECentralPlugin.js

```diff
@@ -1,63 +1,194 @@
+   
 function notSupported() {
     console.log('BLE is not supported on the browser');
 }
 
+
+function getDeviceId(device_id) {
+    return device_id.startsWith('0x') ? parseInt(device_id) : device_id;
+}
+
+
 module.exports = {
+
+    devices: new Map(),
+    scanning: null,
+
     scan: function(services, seconds, success, failure) {
-        notSupported();
-        if (failure) failure();
-    },
-    startScan: function(services, success, failure) {
-        notSupported();
-        if (failure) failure();
+
+        if (!navigator.bluetooth) {
+            failure('Bluetooth is not supported on this browser.');
+            return;
+        }
+
+        this.scanning = true;
+
+        let options = {};
+
+        if (services && services.length) {
+            options.filters = [{
+                services: services.map(getDeviceId)
+            }];
+        } else {
+            options.acceptAllDevices = true;
+        }
+
+        console.log('requestDevice', options);
+        navigator.bluetooth.requestDevice(options).then(device => {
+            if (this.scanning) {  
+                return device.gatt.connect().then(server => {
+                    this.devices.set(device.id, {
+                        device: device,
+                        server: server
+                    })
+                    success(device);
+                });              
+            }
+        }).catch(failure);
     },
     stopScan: function(success, failure) {
-        notSupported();
-        if (failure) failure();
+        this.scanning = false;
+        if (success) success();
     },
     startScanWithOptions: function(services, options, success, failure) {
-        notSupported();
-        if (failure) failure();
+        this.scanning = true;
+        navigator.bluetooth.requestDevice(options).then(device => {
+            if (this.scanning) {                
+                success(device);
+            }
+        }).catch(failure);
     },
     connect: function(device_id, connectSuccess, connectFailure) {
-        notSupported();
-        if (connectFailure) connectFailure();
-    },
-    disconnect: function(device_id, connectSuccess, connectFailure) {
-        notSupported();
-        if (connectFailure) connectFailure();
+        console.log('connect', device_id);
+        if (this.devices.has(device_id)) {
+            connectSuccess(this.devices.get(device_id).device);
+        } else {
+            navigator.bluetooth.requestDevice({filters: [{
+                services: [getDeviceId(device_id)]
```

Comment:
`device_id` is a different concept to a service UUID. I think using it like this makes the interface quite a bit confusing.